client/app/scripts/reducers/root.js

```diff
@@ -421,7 +421,12 @@ export function rootReducer(state = initialState, action) {
       // highlight edge
       state = state.update('highlightedEdgeIds', (highlightedEdgeIds) => {
         highlightedEdgeIds = highlightedEdgeIds.clear();
-        return highlightedEdgeIds.add(action.edgeId);
+        highlightedEdgeIds = highlightedEdgeIds.add(action.edgeId);
+        if (action.bidirectional) {
+          const opposite = action.edgeId.split(EDGE_ID_SEPARATOR).reverse().join(EDGE_ID_SEPARATOR);
+          highlightedEdgeIds = highlightedEdgeIds.add(opposite);
+        }
+        return highlightedEdgeIds;
```

Comment:
I'm looking at how we're updating the highlighted edges list in `ENTER_NODE` action below:

```
// all neighbour combinations because we dont know which direction exists
highlightedEdgeIds = highlightedEdgeIds.union(adjacentNodes.flatMap(adjacentId => [
  [adjacentId, nodeId].join(EDGE_ID_SEPARATOR),
  [nodeId, adjacentId].join(EDGE_ID_SEPARATOR)
]));
```
There we don't care about bidirectionality and we add the reversed edge IDs even when those edges don't exist. Given how `highlightedEdgeIds` is currently used, that shouldn't be a problem. Maybe you can take the same simple approach here and just do something like:

```
highlightedEdgeIds = highlightedEdgeIds.clear();
highlightedEdgeIds = highlightedEdgeIds.add(action.edgeId);

const oppositeEdgeId = action.edgeId.split(EDGE_ID_SEPARATOR).reverse().join(EDGE_ID_SEPARATOR);
return highlightedEdgeIds = highlightedEdgeIds.add(oppositeEdgeId);
```
Then `action.bidirectional` wouldn't be used and you could remove half of the code from this PR. What do you think?